### DISCUSSION TITLE
Reference point to align reprojected coordinates to

### DISCUSSION THREAD

**Discussion by eric542** [corteva/rioxarray#594] [Q&A] [answered]
Hello. I would like to reproject an Xarray.Dataset using `ds.rio.reproject` while ensuring that multiple reprojected datasets result in _aligned_ coordinates. An easy way to achieve this would be to pass on some `align` parameters to the function to provide some reference point that the reprojected x/y coordinates would align to, i.e. the resulting coords would be multiples of the `resolution` parameter (pixel size) which, if interpolated, would align with the reference point. Is this feasible with `rio.reproject` at all?... (e.g. `datacube.Datacube.load` has such a parameter available). Note: I would like to reproject multiple such datasets "in parallel" so I don't have a reference dataset that I could use as a template with `reproject_match`.

**Comment by snowman2** [comment, marked as answer]
It currently isn't possible, but the addition is welcome if someone wanted to add it.

Additional potential helpful references:
- https://odc-geo.readthedocs.io/en/latest/index.html
- https://github.com/corteva/rioxarray/issues/119

**Comment by Kirill888** [comment]
You can use `odc-geo` to compute exact "pixel grid compatible with some other image" that is offset in the world to cover some region with this method:

https://odc-geo.readthedocs.io/en/latest/_api/odc.geo.geobox.GeoBox.enclosing.html

you can then extract shape, crs and transform from that and pass to rioxarray for reprojection, or use reproject built into `odc-geo` (that one is lacking Dask support though)

**Comment by eric542** [comment]
Thanks for the feedback... much appreciated.
